model/minsky.cc

```diff
@@ -713,6 +726,96 @@ namespace minsky
        });
     canvas.requestRedraw();
   }
+  
+  /*
+  For using GSL non-linear least square fitting routines
+  patterned on first Large Nonlinear Least Squares Example.
+  See online documentation at https://www.gnu.org/software/gsl/doc/html/nls.html
+  */
+      
+namespace
+{	
+
+   int residualFunc(const gsl_vector * x, void *params,
+          gsl_vector * f)
+   {
+	if (params==NULL) return GSL_EBADFUNC;
+	try { 	  
+		
+	      auto pars=static_cast<OptParams*>(params);  
+	      
+          int err = GSL_SUCCESS;
+          auto n = ValueVector::stockVars.size();
+          vector<double> xx(n);
+          
+          memcpy(xx.data(),reinterpret_cast<const vector<double>*>(x),n);
+          
+          err=gsl_odeiv2_driver_apply(pars->d->driver, pars->t,numeric_limits<double>::max(),&xx[0]);
+	      
+		  // define residual vector function to be optimized 
+          for (size_t i = 0; i < n; i++)
+              gsl_vector_set (f, i, xx[i] - pars->xi[i]);  
+     }
+     catch  (std::exception& e) {
+		((Minsky*)params)->threadErrMsg=e.what();
+        return GSL_EBADFUNC; 
+	 }   
+	 return  GSL_SUCCESS;      
+                        
+   }
+   
+   static void errHandlerOpt(const char* reason, const char* file, int line, int gsl_errno) {
+         throw error("gsl: %s:%d: %s",file,line,reason);
+   } 
+}      
+
+  int Minsky::optimize(double xf[], OptParams& optPars)
+  {
+    	
+      gsl_set_error_handler(errHandlerOpt);	
+      const gsl_multifit_nlinear_type *T = gsl_multifit_nlinear_trust;
+      gsl_multifit_nlinear_workspace *w;
```

Comment:
I don't see anywhere where this variable is initialised

Reply:
Ah found it. Please move the declaration to line 891 where it is initialised.

Reply:
Done.